### DISCUSSION TITLE
Is it possible to work with typescript?

### DISCUSSION THREAD

**Discussion by joe-7peaks** [originjs/vite-plugin-federation#477] [Q&A]
I'm trying to setup the project for PoC reason and has issue with typescript. So, I'm just curious if it's possible to work with typescript project?

here is my code
**remote - vite.config.ts**
```import { defineConfig } from 'vite'
import federation from "@originjs/vite-plugin-federation";
import react from '@vitejs/plugin-react'

// https://vitejs.dev/config/
export default defineConfig({
  plugins: [
    federation({
      name: 'remote-app',
      filename: 'remoteEntry.js',
      // Modules to expose
      exposes: {
          './Button': './src/components/Button.tsx',
      }
    }),
    react()
  ],
})
```

**host - vite.config.ts**
```import { defineConfig } from 'vite'
import react from '@vitejs/plugin-react'
import federation from "@originjs/vite-plugin-federation";

// https://vitejs.dev/config/
export default defineConfig({
  plugins: [
    federation({
      name: 'host-app',
      remotes: {
          remote_app: "http://localhost:4173//assets/remoteEntry.js",
      },
    }),
    react()
  ],
})
```

host App.tsx
```import './App.css'
import Button from 'remote_app/Button';

function App() {

  return (
    <Button />
  )
}

export default App
```

Anyway, I got Cannot find module 'remote_app/Button' or its corresponding type declarations.ts(2307).

**Comment by dk-v** [comment]
You can fix the typescript error by adding a declaration file at the root of your project like this:
`declare module 'remote_App/Button';`
This should make the error disappear.

**Reply by PaRoxUs** [reply]
But how to share types between the projects in the most sustainable way?

**Reply by DevShoesed** [reply]
It's possible to use the plugin for native federation  typescript ? https://github.com/module-federation/universe/tree/main/packages/native-federation-typescript
